pkg/common/cns-lib/vsphere/utils.go

```diff
@@ -469,3 +470,33 @@ func IsvSphereVersion70U3orAbove(ctx context.Context, aboutInfo types.AboutInfo)
 	// For all other versions.
 	return false, nil
 }
+
+// IsVolumeCreationSuspended checks whether a given Datastore has cns.vmware.com/datastoreSuspended customValue
+func IsVolumeCreationSuspended(ctx context.Context, datastoreInfo *DatastoreInfo) bool {
+	log := logger.GetLogger(ctx)
+
+	for _, customValField := range datastoreInfo.CustomValues {
+		customVal := customValField.(*types.CustomFieldStringValue)
+		if customVal.Value == cnsMgrDatastoreSuspended {
+			log.Debugf("Ignoring datastore %v as it is suspended.", datastoreInfo.Datastore)
```

Comment:
can we print both ds name and dsmoref as Info level log for suspended datastores?
when failure happens SRE can determine this from the logs.